doc/source/dev/development_environment.rst

```diff
@@ -203,3 +203,19 @@ typically packaged as ``python-dbg``) is highly recommended.
 .. _virtualenvwrapper: http://www.doughellmann.com/projects/virtualenvwrapper/
 
 .. _Waf: https://code.google.com/p/waf/
+
+Understanding the code & getting started
+----------------------------------------
+
+The best strategy to better understand the code base is to pick something you 
+want to change and start reading the code to figure out how it works. 
+When you're in doubt, you can ask questions on the mailing list or on the 
+NumPy Github page. It is perfectly okay if your pull requests aren't perfect, 
```

Comment:
Hmm, the github page is for PR and bug reports, guestions should go to the mailing list.